spec/models/step_by_step_page_spec.rb

```diff
@@ -245,4 +245,25 @@
       end
     end
   end
+
+  describe '.discard_notes' do
+    before(:each) do
+      @step_by_step_page = create(:step_by_step_page)
+      create(:internal_change_note, edition_number: 1, step_by_step_page_id: @step_by_step_page.id)
+      create(:internal_change_note, step_by_step_page_id: @step_by_step_page.id)
+    end
+    context 'when there are existing change notes with a version and new change notes without a version' do
+      before(:each) do
+        @step_by_step_page.discard_notes
+        @step_by_step_page.reload
+      end
+      # this is necessarily a longer and more complex query because it's searching for a negative
+      it 'should delete the change notes without an edition' do
```

Comment:
I think you can combine these two tests into one, "it should only delete the change notes without an edition"
and then check that:
1. @step_by_step_page.internal_change_notes.count is 1
2. @step_by_step_page.internal_change_notes.first[:edition_number] is also 1

Reply:
I don't understand the second test. Is it testing to make sure that the right edition number is allocated to the only change note?